realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -560,7 +560,8 @@ void BaseRealSenseNode::frame_callback(rs2::frame frame)
                 frame_to_send = original_depth_frame;
             publishFrame(frame_to_send, t, DEPTH, _images, _info_publishers, _image_publishers);
 
-            if(_enable_rgbd)
+            // publish RGBD only if rgbd enabled and both depth and color frames exist
+            if(_enable_rgbd && original_color_frame)
```

Comment:
notice line 554, which makes sure we already got a depth frame (original_depth_frame != null)

Reply:
I would write it in the comment :)
`// publish RGBD only if rgbd enabled and both depth and color frames exist (on this line we already know original_depth_frame  is valid)`

Reply:
Updated.
Anyway, all of these parts are scheduled for redesign in the main branch :)